harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -363,158 +808,95 @@ def _auto_step_lr_scheduler_per_batch(
-
-        # Ignore batch_metrics entirely for custom reducers; there's no guarantee that per-batch
-        # metrics are even logical for a custom reducer.
-        with self.prof.record_timing("reduce_metrics"):
-            metrics["avg_metrics"].update(
-                pytorch._convert_metrics_to_numpy(self.context.reduce_metrics(for_training=True))
+        with contextlib.ExitStack() as exit_stack:
+            exit_stack.enter_context(self.prof.record_timing("train_batch", requires_sync=False))
+            if self.context.profiler:
+                exit_stack.enter_context(self.context.profiler)
+
+            training_metrics = self.trial.train_batch(
+                batch=batch,
+                epoch_idx=epoch_idx,
+                batch_idx=batch_idx,
             )
 
-        if not self.is_chief:
-            # The training metrics are reported only in the chief process.
-            return {}
+            if self.context.profiler:
+                self.context.profiler.step()
 
-        step_duration = time.time() - step_start_time
-        logging.info(det.util.make_timing_log("trained", step_duration, num_inputs, num_batches))
-        self.metric_writer.on_train_step_end(
-            self.steps_completed,
-            metrics["avg_metrics"],
-            metrics["batch_metrics"],
+        if self._should_update_scaler():
+            # We update the scaler once after train_batch is done because the GradScaler is
+            # expected to be one-per-training-loop, with one .update() call after all .step(opt)
+            # calls for that batch are completed [1].
+            #
+            # [1] pytorch.org/docs/master/notes/amp_examples.html
+            #         #working-with-multiple-models-losses-and-optimizers
+            self.context._scaler.update()  # type: ignore
+
+        if isinstance(training_metrics, torch.Tensor):
+            training_metrics = {"loss": training_metrics}
+
+        # Step learning rate of a pytorch.LRScheduler.
+        with self.prof.record_timing("step_lr_schedulers"):
+            for lr_scheduler in self.context.lr_schedulers:
+                self._auto_step_lr_scheduler_per_batch(batch_idx, lr_scheduler)
+
+        with self.prof.record_timing("from_device"):
+            for name, metric in training_metrics.items():
+                # Convert PyTorch metric values to NumPy, so that
+                # `det.util.encode_json` handles them properly without
+                # needing a dependency on PyTorch.
+                if isinstance(metric, torch.Tensor):
+                    metric = metric.cpu().detach().numpy()
+                training_metrics[name] = metric
+
+        batch_dur = time.time() - batch_start_time
+        samples_per_second = self.trial.get_batch_length(batch) / batch_dur
+        samples_per_second *= self.context.distributed.size
+        self.prof.record_metric("samples_per_second", samples_per_second)
+
+        common.check.is_instance(
```

Comment:
did you change this from an if statement to use the deprecated `check` module?  Can you change it back, and not use `check`?  And if you see them and you're changing nearby code, delete them please.